pkg/manager/internal.go

```diff
@@ -150,6 +154,45 @@ type controllerManager struct {
 	retryPeriod time.Duration
 }
 
+type errSignaler struct {
+	// errSignal indicates that an error occurred, when closed.  It shouldn't
+	// be written to.
+	errSignal chan struct{}
+
+	// err is the received error
+	err error
+
+	mu sync.Mutex
+}
+
+func (r *errSignaler) SignalError(err error) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
```

Comment:
we should handle if this gets called with `nil` err. That can lead to calling `close` on a closed channel and cause panic.

Reply:
fixed